docs/api/json.md

```diff
@@ -0,0 +1,291 @@
+- `SS` is seconds from `00-59`.
+- `fffffffff` is the sub-second precision up to 9 digits from `0` to `999999999`
+
+A leading 0 can be skipped in hours, minutes and seconds. Specifying the
+subsecond precision is optional and can have any number of digits from 0 to 9.
+Leading and trailing whitespaces are ignored.
+
+### `DATE`
+
+Specifies dates using the `YYYY-MM-DD` format.
+
+- `YYYY` is the year from `0001-9999`
+- `MM` is the month from `01-12`
+- `DD` is the day from `01-31`
+
+Invalid dates (e.g., `1997-02-29`) are rejected with an error during ingress.
+Leading zeros are skipped, e.g., `0001-1-01`, `1-1-1`, `0000-1-1` are all
+equal and valid. Leading and trailing whitespaces are ignored.
+
+### `TIMESTAMP`
+
+Specifies dates using the `YYYY-MM-DD HH:MM:SS.fff` format.
+
+- `YYYY` is the year from `0001-9999`
+- `MM` is the month from `01-12`
+- `DD` is the day from `01-31`
+- `HH` is hours from `00-23`.
+- `MM` is minutes from `00-59`.
+- `SS` is seconds from `00-59`.
+- `fff` is the sub-second precision up to 3 digits from `0` to `999`
+
+Note that the same rules as specified in the Date and Time sections apply,
+except that the sub-second precision is limited to three digits (microseconds).
+Specifying more digits for the subsecond precision on ingress will trim the
+fraction to microseconds. Leading and trailing whitespaces are ignored.
+
+### `ARRAY`
+
+Arrays are encoded as JSON arrays.
+
+## Encoding data change events
+
+Feldera operates over streams of **data change events**.
+A data change event represents an insertion, deletion, or modification of a
+single row in a SQL table or view.  We currently support two
+data change event formats in JSON: (1) the raw format and (2) the insert/delete format.
+
+### The raw format
+
+This format is applicable to append-only event streams where rows can only
+be inserted but not deleted.  In this case, a data change event can be
+represented as a SQL record without any additional framing, with the insert
+operation being implied.
+
+### The insert/delete format
+
+A data change event in this format is represented as a JSON object with a
+single key, which must be equal to `insert` or `delete`.  The associated value
+represents a table row to be inserted or deleted, encoded using the format
+documented above, e.g.:
+
+```json
+{"delete": {"part": 1, "vendor": 2, "price": 10000}}
+{"insert": {"part": 1, "vendor": 2, "price": 30000}}
+```
+
+## Encoding multiple changes
+
+Data change events are exchanged as data streams transmitted over transports such
+as HTTP or Kafka.  We use newline-delimited JSON (NDJSON) to assemble individual
+data change events into a stream.  In NDJSON, each line within the data stream
+represents a self-contained and valid JSON document, in this case, a data change
+event:
+
+```json
+{"delete": {"part": 1, "vendor": 2, "price": 10000}}
+{"insert": {"part": 1, "vendor": 2, "price": 30000}}
+{"insert": {"part": 2, "vendor": 3, "price": 34000}}
+...
```

Comment:
maybe put the ... as // ... to make the json parser happy

Reply:
yeah, this is not valid json, but docusaurus handles it ok.